File: server/src/schema/database/import.ts
import {Import} from "../../worker/import/import";

export interface ImportRecord {
  id: number;
  profileId: number;
  year: number;

  userId: number;

  created: Date;
  started?: Date;
  finished?: Date;

  status: 'pending' | 'processing' | 'success' | 'error';
  error?: string;

  validity?: string;
  append: boolean;
  logs?: string;
  format: Import.Format;
  importDir: string;
}

File: importer/Cityvizor.Importer/Cityvizor.Importer.Infrastructure/Entities/Import.cs
﻿using Cityvizor.Importer.Infrastructure.Enums;

namespace Cityvizor.Importer.Infrastructure.Entities;
public class Import
{
    public int Id { get; set; }
    public int ProfileId { get; set; }
    public int Year { get; set; }
    public int? UserId { get; set; }
    public DateTime Created { get; set; }
    public DateTime? Started { get; set; }
    public DateTime? Finished { get; set; }
    public ImportStatus Status { get; set; }
    public string? Error { get; set; }
    public DateTime? Validity { get; set; }
    public bool Append {  get; set; }
    public string? Logs { get; set; }
}
